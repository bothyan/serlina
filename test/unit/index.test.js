const { Serlina } = require('../../dist')
const path = require('path')

const GLOBAL_PAYLOAD = { foo: 'foo' }
const PAGE1 = 'page1' // don't have style
const PAGE2 = 'page2' // have style

describe('Serlina', () => {

  let app

  beforeAll(async () => {
    app = new Serlina({
      baseDir: path.resolve(__dirname, './fixtures/sample'),
      __testing: true
    })
    await app.prepare()
    app.inject(GLOBAL_PAYLOAD)
  })

  test('snapshot', async () => {
    const rendered = await app.render(PAGE1)
    expect(rendered.string).toMatchSnapshot()
    const rendered2 = await app.render(PAGE2)
    expect(rendered2.string).toMatchSnapshot()
  })

  test('inject', async () => {
    const payload = {
      bar: 'bar'
    }
    const rendered = await app.render(PAGE1,payload)
    expect(rendered.__injected).toEqual(Object.assign({},GLOBAL_PAYLOAD, payload))
  })

  test('page assets', async () => {
    const rendered = await app.render(PAGE1)
    expect(rendered.__pageScripts).toEqual([
      PAGE1 + '.js',
      '_SERLINA_VENDOR.js',
      '_SERLINA_MAIN.js'
    ])

    expect(rendered.__pageStyles).toEqual([])
  })

  test('page assets (have style)', async () => {
    const rendered = await app.render(PAGE2)
    expect(rendered.__pageScripts).toEqual([
      PAGE2 + '.js',
      '_SERLINA_VENDOR.js',
      '_SERLINA_MAIN.js'
    ])

    expect(rendered.__pageStyles).toEqual([
      PAGE2 + '.css'
    ])
  })

  test('render a non exist page', async () => {
    const rendered = await app.render('non-exist')

    expect(rendered.__pageScripts).toEqual([
      '_404.js',
      '_SERLINA_VENDOR.js',
      '_SERLINA_MAIN.js'
    ])
  })
})