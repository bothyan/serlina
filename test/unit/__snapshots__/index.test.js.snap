// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Serlina snapshot 1`] = `"<!DOCTYPE html><html data-reactroot=\\"\\"><head><title data-react-helmet=\\"true\\"></title><script src=\\"http://127.0.0.1:3000/_SERLINA_VENDOR.js\\"></script><script>window.__serlina__DATA={};window.__serlina__DATA.pageInitialProps={};</script></head><body><div id=\\"app\\"><div data-reactroot=\\"\\">Page1</div></div><script src=\\"http://127.0.0.1:3000/page1.js\\"></script><script src=\\"http://127.0.0.1:3000/_SERLINA_MAIN.js\\"></script></body></html>"`;

exports[`Serlina snapshot 2`] = `"<!DOCTYPE html><html data-reactroot=\\"\\"><head><title data-react-helmet=\\"true\\"></title><link rel=\\"stylesheet\\" href=\\"http://127.0.0.1:3000/page2.css\\"/><script src=\\"http://127.0.0.1:3000/_SERLINA_VENDOR.js\\"></script><script>window.__serlina__DATA={};window.__serlina__DATA.pageInitialProps={};</script></head><body><div id=\\"app\\"><div data-reactroot=\\"\\">Page2</div></div><script src=\\"http://127.0.0.1:3000/page2.js\\"></script><script src=\\"http://127.0.0.1:3000/_SERLINA_MAIN.js\\"></script></body></html>"`;
